src/nvim/eval.c

```diff
@@ -7058,6 +7061,11 @@ char *set_cmdarg(exarg_T *eap, char *oldarg)
     snprintf(newval + strlen(newval), newval_len, " ++bad=%c",
              eap->bad_char);
   }
+
+  if (eap->mkdir_p) {
+    STRCAT(newval, " ++p");
```

Comment:
```suggestion
    strcat(newval, " ++p");
```

Reply:
I used `STRCAT` here instead of `strcat` to keep consistency in the code since that macro was previously used for the ++edit option as well

Reply:
if STRCAT is used where it isn't needed that's just code that hasn't been cleaned up. We don't want to repeat that pattern in new code :)

Reply:
Changing STRNCMP to strncmp didn't cause any trouble, but the lintc check fails for strcat.